staging/src/k8s.io/apiserver/pkg/endpoints/filters/resource_version.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filters
+
+import (
+	"errors"
+	"net/http"
+	"strconv"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/endpoints/handlers/responsewriters"
+	"k8s.io/apiserver/pkg/endpoints/request"
+)
+
+// WithResourceVersionValidation ensures that the "resource version" query prameter
+// if specified is valid.
+// If an invalid value of resourceVersion is specified in the request URI, the
+// request is rejected with a 400.
+func WithResourceVersionValidation(handler http.Handler, negotiatedSerializer runtime.NegotiatedSerializer) http.Handler {
```

Comment:
The field is a string, all strings are valid from a schema perspective. Not all requests have a resource version parameter, so we should not use a filter to accomplish this anyway.

Reply:
yeah, usually the validation happens in [the handlers ](https://github.com/kubernetes/kubernetes/blob/5d61e18a4343773105e39c36b944b4f223e70eaf/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/get.go#L203)or even directly in [the registry](https://github.com/kubernetes/kubernetes/blob/8b98305858b107369f2c9b9fd8ef1c5b0da078c0/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L819). We should have something similar for checking the `resourceVersion`.